libsolidity/codegen/ABIFunctions.h

```diff
@@ -75,10 +93,29 @@ class ABIFunctions
 	/// Returns a pointer to the end of the area written in memory.
 	/// Does not allocate memory (does not change the free memory pointer), but writes
 	/// to memory starting at $headStart and an unrestricted amount after that.
-	std::string tupleEncoder(
+	std::string tupleEncoderReversed(
 		TypePointers const& _givenTypes,
 		TypePointers const& _targetTypes,
 		bool _encodeAsLibraryTypes = false
+	) {
+		return tupleEncoder(_givenTypes, _targetTypes, _encodeAsLibraryTypes, true);
+	}
+
+	/// @returns name of an assembly function to encode values of @a _givenTypes
+	/// with packed encoding into memory, converting the types to @a _targetTypes on the fly.
+	/// Parameters are: <memPos> <value_1> ... <value_n>, i.e.
+	/// the layout on the stack is <value_n> ... <value_1> <memPos> with
+	/// the top of the stack on the right.
```

Comment:
Apart from these three lines, we could probably just refer to ``@see tupleEncoder``... but maybe duplicating is not too bad...